testing/modelmanager_test.go

```diff
@@ -50,12 +66,772 @@ func (s *modelE2eManagerSuite) TestListModelSummaries(c *gc.C) {
+				UUID:  s.Model2.UUID.String,
+				Owner: "charlie@canonical.com",
+				Type:  "iaas",
+			}, {
+				Name:  s.Model3.Name,
+				UUID:  s.Model3.UUID.String,
+				Owner: "charlie@canonical.com",
+				Type:  "iaas",
+			},
+		},
+	)
+}
+
+func (s *modelE2EManagerSuite) TestModelInfo(c *gc.C) {
+	model4Name := petname.Generate(2, "-")
+	mt4 := s.AddModel(c, names.NewUserTag("charlie@canonical.com"), model4Name, names.NewCloudTag(jimmtest.TestE2ECloudName), jimmtest.TestE2ECloudRegionName, s.Model2.CloudCredential.ResourceTag())
+
+	bobIdentity, err := dbmodel.NewIdentity("bob@canonical.com")
+	c.Assert(err, gc.IsNil)
+	bob := openfga.NewUser(bobIdentity, s.OFGAClient)
+	err = bob.SetModelAccess(context.Background(), mt4, ofganames.WriterRelation)
+	c.Assert(err, gc.Equals, nil)
+
+	model5Name := petname.Generate(2, "-")
+	mt5 := s.AddModel(c, names.NewUserTag("charlie@canonical.com"), model5Name, names.NewCloudTag(jimmtest.TestE2ECloudName), jimmtest.TestE2ECloudRegionName, s.Model2.CloudCredential.ResourceTag())
+	// TODO (alesstimec) change once granting has been re-implemented
+	// err = client.GrantModel("bob@canonical.com", "admin", mt5.Id())
+	// c.Assert(err, gc.Equals, nil)
+	err = bob.SetModelAccess(context.Background(), mt5, ofganames.AdministratorRelation)
+	c.Assert(err, gc.Equals, nil)
+
+	conn := s.Open(c, nil, "bob", nil)
+	defer conn.Close()
+	client := modelmanager.NewClient(conn)
+	models, err := client.ModelInfo([]names.ModelTag{
+		s.Model.ResourceTag(),
+		s.Model2.ResourceTag(),
+		s.Model3.ResourceTag(),
+		mt4,
+		mt5,
+		names.NewModelTag("00000000-0000-0000-0000-000000000007"),
+	})
+	c.Assert(err, gc.Equals, nil)
+
+	// Verify we got results for all 6 model requests
+	c.Assert(models, gc.HasLen, 6)
+
+	// Helper to find user access in Users slice
+	findUserAccess := func(users []jujuparams.ModelUserInfo, username string) string {
+		for _, u := range users {
+			if u.UserName == username {
+				return string(u.Access)
+			}
+		}
+		return ""
+	}
+
+	// Model 1 (s.Model) - bob has admin access
+	c.Assert(models[0].Error, gc.IsNil)
+	c.Assert(models[0].Result.Name, gc.Equals, s.Model.Name)
+	c.Assert(models[0].Result.UUID, gc.Equals, s.Model.UUID.String)
+	c.Assert(models[0].Result.Life, gc.Equals, life.Alive)
+	c.Assert(models[0].Result.ControllerUUID, gc.Equals, jimmtest.ControllerUUID)
+	c.Assert(models[0].Result.CloudCredentialTag, gc.Equals, s.Model.CloudCredential.Tag().String())
+	c.Assert(models[0].Result.OwnerTag, gc.Equals, names.NewUserTag("bob@canonical.com").String())
+	// As admin, bob can see all users
+	c.Assert(findUserAccess(models[0].Result.Users, "bob@canonical.com"), gc.Equals, "admin")
+	c.Assert(findUserAccess(models[0].Result.Users, "alice@canonical.com"), gc.Equals, "admin")
+
+	// Model 2 (s.Model2) - bob has no access
+	c.Assert(models[1].Result, gc.IsNil)
+	c.Assert(models[1].Error, gc.Not(gc.IsNil))
+	c.Assert(models[1].Error.Code, gc.Equals, jujuparams.CodeUnauthorized)
```

Comment:
It's a bit strange we return models where the user doesn't have access.
Ah but that's because we specifically asked for it, okay that's fine.